src/main/java/picard/fingerprint/FingerprintChecker.java

```diff
@@ -672,27 +684,19 @@ public Map<FingerprintIdDetails, Fingerprint> fingerprintFiles(final Collection<
                 final Map<FingerprintIdDetails, Fingerprint> oneFileFingerprints;
                 log.debug("Processed file: " + p.toUri().toString() + " (" + filesRead.get() + ")");
 
+                // Determine whether valid index path given for this file
+                final Path indexPath = (indexPathMap != null && indexPathMap.containsKey(p)) ? indexPathMap.get(p) : null;
```

Comment:
Don't need to check that p is in indexPathMap. ` indexPathMap.get(p)` will return null if it can't find p in the map

Reply:
Nice, removed the second boolean expression for the ternary check